Configuration/HLT/python/addOnTestsHLT.py

```diff
@@ -22,25 +22,25 @@ def addOnTestsHLT():
                     'hlt_mc_PRef' : ['cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run3_mc_PRef --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run3 --fileout file:RelVal_Raw_PRef_MC.root', 
                                'HLTrigger/Configuration/test/OnLine_HLT_PRef.py',
                                'cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root'],
-                    'hlt_data_Fake' : ['cmsDriver.py RelVal -s L1REPACK:GT1   --data --scenario=pp -n 10 --conditions auto:run1_hlt_Fake --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_DATA.root --filein /store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root',
+                    'hlt_data_Fake' : ['cmsDriver.py RelVal -s L1REPACK:GT1   --data --scenario=pp -n 10 --conditions auto:run1_hlt_Fake --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_DATA.root --filein root://eoscms.cern.ch//eos/cms/store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/RAW/Run2012A_MuEG_run191718/14932935-E289-E111-830C-5404A6388697.root',
```

Comment:
As far as I can tell, all the files in the removed lines (in both files of this PR) are already cached by the IB system. @smuzaffar Do we have preference between the IB cache and explicit EOS group area?

Reply:
As mentioned in https://github.com/cms-sw/cmssw/issues/40013#issue-1439703206 [**], an even better solution would be to figure out how to extend the caching mechanism to the files used by the HLT-Validation tests (it might be simple, but I don't know how), without relying on having them in the TSG area.

> [**] It might be useful to know from experts if there are ways to cache these files similarly to what is done for the EDM files used in RelVal wfs and addOnTests.

Reply:
bot caches all data files read by runTheMatrix workflows, addOns and unit tests. It parses the log files and search for messages like 
```
09-Nov-2022 16:05:53 CET  Initiating request to open file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_12_5_0_pre4/RelValTTbar_14TeV/GEN-SIM/124X_mcRun3_2022_realistic_v10_BS2022-v1/10000/597c0099-c758-4af7-84c3-02b6b15de660.root
```
We can extend bot for parse the HLT-Validation logs too and cache those files

Reply:
Still trying to understand, so sorry if the questions are basic.

 - For the HLT-Validation tests that run in IBs, if we were to only specify `/store/{mc,data}/[..]` in [`cmsDriver.csh`](https://github.com/cms-sw/cmssw/blob/0c2b667757069591869eae6b56dcc530726c104c/HLTrigger/Configuration/test/cmsDriver.csh#L34), would these tests be redirected to the cmsbot cache when `cmsRun` runs? How does that work?

 - One problem is that we also often run these HLT-Val tests locally, and in that case it is convenient to just point to the [TSG EOS area](https://github.com/cms-sw/cmssw/blob/0c2b667757069591869eae6b56dcc530726c104c/HLTrigger/Configuration/test/cmsDriver.csh#L43) to avoid any issues with file access. Is there a way to run a script like this `cmsDriver.csh` interactively in a way that it eventually redirects to the cmsbot cache?

Reply:
@missirol 
> - For the HLT-Validation tests that run in IBs, if we were to only specify /store/{mc,data}/[..] in [cmsDriver.csh](https://github.com/cms-sw/cmssw/blob/0c2b667757069591869eae6b56dcc530726c104c/HLTrigger/Configuration/test/cmsDriver.csh#L34), would these tests be redirected to the cmsbot cache when cmsRun runs? How does that work?

For now HLT tests are not using the bot cache but it is trivial to update these tests to start using bot eos cache. We just need to set [a] for this job ( which I just did https://github.com/cms-sw/cms-bot/commit/5838fb2ca2f90522be47239b1a60c8ffe333522a ). `SITECONF` in cms-ib area has special rules [b] to first look for bot cache and then fallback to rxootd global re-director.

> One problem is that we also often run these HLT-Val tests locally, and in that case it is convenient to just point to the [TSG EOS area](https://github.com/cms-sw/cmssw/blob/0c2b667757069591869eae6b56dcc530726c104c/HLTrigger/Configuration/test/cmsDriver.csh#L43) to avoid any issues with file access. Is there a way to run a script like this cmsDriver.csh interactively in a way that it eventually redirects to the cmsbot cache?

If you have access to cvmfs then you can set [a] locally and then your tests should first look in to bot cache.

[a]
```
export CMS_PATH="/cvmfs/cms-ib.cern.ch"
export SITECONFIG_PATH="/cvmfs/cms-ib.cern.ch/SITECONF/local"
```

[b]
https://github.com/cms-sw/siteconf/blob/master/local/PhEDEx/storage.xml#L9-L11
https://github.com/cms-sw/siteconf/blob/master/local/storage.json#L22-L41

Reply:
@smuzaffar , I'm looking into updating the HLT-Val tests to use the bot cache. I think the Data files are all in the cache (as Matti noted); for the MC files, that's not the case.

Those tests use GEN-SIM files ([example](https://github.com/cms-sw/cmssw/blob/CMSSW_12_1_X/HLTrigger/Configuration/test/cmsDriver.csh#L15-L26)) that I can't find anywhere (old datasets), except for the copies we do have in the TSG EOS area. Would it be possible to copy them manually into the bot cache (something like [1])?

Regarding

> We can extend bot for parse the HLT-Validation logs too and cache those files

Is this something you could do in the next days? Should I wait for that before updating these PRs?

[1]
```bash
#!/bin/bash

copyFile(){
  local INPFILE=/eos/cms"${1}"
  local OUTFILE=/eos/cms/store/user/cmsbuild"${2}"
  ([ -f "${INPFILE}" ] && [ ! -f "${OUTFILE}" ]) || return 0
  mkdir -p "$(dirname ${OUTFILE})"
  cp "${INPFILE}" "${OUTFILE}"
}

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_5/143C21CD-E8A2-E311-87BE-0025904C66E8.root \
/store/relval/CMSSW_5_3_16/RelValPyquen_ZeemumuJets_pt10_2760GeV/GEN-SIM/PU_STARTHI53_LV1_mar03-v2/00000/143C21CD-E8A2-E311-87BE-0025904C66E8.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_5/DE03BB7E-F429-E211-A0B4-001A928116CC.root \
/store/relval/CMSSW_5_3_6-START53_V14/RelValProdTTbar/GEN-SIM/v2/00000/DE03BB7E-F429-E211-A0B4-001A928116CC.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_8/06A6C86B-C634-E611-93A5-0CC47A74525A.root \
/store/relval/CMSSW_8_0_11/RelValProdTTbar/GEN-SIM/80X_mcRun1_realistic_v4-v1/10000/06A6C86B-C634-E611-93A5-0CC47A74525A.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_8/06F2C3AC-8957-E611-9DDF-0025905B85D8.root \
/store/relval/CMSSW_8_0_16/RelValProdTTbar_13/GEN-SIM/80X_mcRun2_asymptotic_v16_gs7120p2-v1/10000/06F2C3AC-8957-E611-9DDF-0025905B85D8.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_8/F8FC5F64-1657-E611-A57E-002590A887F0.root \
/store/relval/CMSSW_8_0_16/RelValZEEMM_13_HI/GEN-SIM/80X_mcRun2_HeavyIon_v9-v1/10000/F8FC5F64-1657-E611-A57E-002590A887F0.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_9_phase1/14F749AC-8AFE-E611-9821-0CC47A78A4A0.root \
/store/relval/CMSSW_9_0_0_pre5/RelValTTbar_13/GEN-SIM/90X_upgrade2017_realistic_v15-v1/00000/14F749AC-8AFE-E611-9821-0CC47A78A4A0.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_10/E288668E-A2D1-D446-A401-D71EA43DD796.root \
/store/relval/CMSSW_10_3_0_pre5/RelValZEEMM_13_HI/GEN-SIM/103X_upgrade2018_realistic_v7-v1/10000/E288668E-A2D1-D446-A401-D71EA43DD796.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_11/3ee9ba1e-0ef8-4242-8343-cff886c9f7b3.root \
/store/relval/CMSSW_11_2_0_pre8/RelValTTbar_13/GEN-SIM/112X_mcRun3_2021_design_v10-v1/00000/3ee9ba1e-0ef8-4242-8343-cff886c9f7b3.root

copyFile /store/group/dpg_trigger/comm_trigger/TriggerStudiesGroup/STORM/GEN-SIM/CMSSW_11/65e018bc-2a25-4f53-b9cf-aba35a7b212d.root \
/store/relval/CMSSW_11_2_0_pre8/RelValZEE_14_HI_2021/GEN-SIM/112X_mcRun3_2021_realistic_HI_v11-v1/00000/65e018bc-2a25-4f53-b9cf-aba35a7b212d.root
```

Reply:
@missirol , all these files are now available in ibeos cache

Reply:
thanks @smuzaffar !